core/app/helpers/spree/products_helper.rb

```diff
@@ -71,6 +95,33 @@ def available_status(product) # will return a human readable string
       end
     end
 
+    def product_images(product, variants)
+      variants = if product.variants_and_option_values(current_currency).any?
+        variants.reject(&:is_master)
+      else
+        variants
+      end
+
+      variants.map(&:images).flatten
+    end
+
+    def product_variants_matrix
+      Spree::VariantPresenter.new(@variants).call.to_json
+    end
+
+    def related_products
```

Comment:
@KirillKayumov we need to add a check here

```suggestion
    def related_products
      return [] unless @product.defined?(:related_products)
      
```

`related_products` method is provided by https://github.com/spree-contrib/spree_related_products extension it doesn't exist in spree core codebase